pkg/apis/db/v1alpha1/cassandracluster_types_test.go

```diff
@@ -457,3 +458,45 @@ func TestComputeLastAppliedConfiguration(t *testing.T) {
 	assert.Equal(result, string(lastAppliedConfiguration))
 
 }
+
+func TestSetDefaults(t *testing.T) {
+	assert := assert.New(t)
+
+	cluster := CassandraCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CassandraCluster",
+			APIVersion: "db.orange.com/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "defaults-test",
+			Namespace: "default",
+			Labels:    map[string]string{"cluster": "k8s.pic"},
+		},
+		Spec: CassandraClusterSpec{
+			Resources: CassandraResources{
+				Requests: CPUAndMem{
+					CPU: "500m",
+					Memory: "1Gi",
+				},
+			},
+			Topology: Topology{
+				DC: DCSlice{
+					DC{
+						Name: "dc1",
+						Rack: RackSlice{
+							Rack{
+								Name: "rack1",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	cluster.SetDefaults()
+
+	assert.Equal("orangeopensource/cassandra-image", cluster.Spec.BaseImage)
+	assert.Equal(CPUAndMem{CPU: "500m", Memory: "1Gi"}, cluster.Spec.Resources.Limits)
+}
```

Comment:
Let’s add tests for the other default parameters that are set, wdyt ?

Reply:
>Let’s add tests for the other default parameters that are set, wdyt ?

👍 Absolutely!

I actually spent more time looking into validation, but will that be addressed with #34?

Reply:
@cscetbon I update `TestSetDefaults`.  Do you want to address validation in a separate PR for #34? If so, is there anything else needed for this PR?

Reply:
Yes, no need to spend time here on validation. Thanks for your contribution !